### PR TITLE
Fetch users budgets API

### LINKED ISSUE #107: Fetch budgets API
- `GET /users/:userId/budgets`

### PR DESCRIPTION
**Pull request by dlarocque**
- `GET /users/:userId/budgets` 
- Returns a list of the users budgets, and the `totalExpense`- the total expenses that have gone towards the budget so far
- Resets on first day of week, month, and year

Closes #107

### CODE REVIEW

**Review by BarbzCodez — APPROVED** [review]
image

File: server/src/routes/auth.ts [hunk 1]
```diff
@@ -464,4 +465,54 @@ router.get(
...
+        }),
+      );
+
+      res.status(200).json({ success: true, data: budgetWithExpenses });
+    } catch (error) {
+      res.status(500).json({ message: 'Server error' });
+    }
+  },
+);
+
```
Comment (BarbzCodez) [inline server/src/routes/auth.ts:517]: why is this under auth route (?)

Comment (BarbzCodez) [reply server/src/routes/auth.ts:517]: nvm makes sense bc you are fetching for that user

**Review by DucharmeEthan — APPROVED** [review]
Looks good to me!

**Pull request merged**